File: Cap xA/Capitulo 07 - Exercicio 1a/c07ex1a.lua
--[[
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: Apendice A - CORRECAO DE ALGUNS EXERCICIOS DE FIXACAO
 *
 * Programa ..: LISTA CLASSIFICADA POR NUMERO - DECRESCENTE
--]]

local A = {}
print("CLASSIFICACAO (DECRESCENTE)\n")
for I = 1, 12, 1 do
  io.write(string.format("Entre o %2do. numero: ", I))
  A[I] = tonumber(io.read())
end
for I = 1, 11, 1 do
  for J = I + 1, 12, 1 do
    if (A[I] < A[J]) then
      X = A[I]
      A[I] = A[J]
      A[J] = X
    end
  end
end
print("\nNUMEROS CLASSIFICADOS\n")
for I = 1, 12, 1 do
  print(A[I])
end

File: Cap xA/Capitulo 07 - Exercicio 1a/c07ex1a.rb
=begin
  Livro .....: Algoritmos
               Logica para Desenvolvimento de Programacao de
               Computadores
  Autores ...: Jose Augusto N. G. Manzano
               Jayr Figueiredo de Oliveira
  Editora ...: Erica
  
  Capitulo ..: Apendice A - CORRECAO DE ALGUNS EXERCICIOS DE FIXACAO
  
  Programa ..: LISTA CLASSIFICADA POR NUMERO - DECRESCENTE
=end

puts 'CLASSIFICACAO (DECRESCENTE)'
puts

A = Array.new(12)

(1..12).each do |i|
  print "Entre o #{sprintf('%2d', i)}o. numero: "
  A[i] = gets.chomp.to_i
end

(1..11).each do |i|
  (i + 1..12).each do |j|
    if A[i] < A[j]
      x = A[i]
      A[i] = A[j]
      A[j] = x
    end
  end
end

puts
puts 'NUMEROS CLASSIFICADOS'
puts

(1..12).each do |i|
  puts A[i]
end
